docs/build/run_a_pipeline.md

```diff
@@ -62,36 +62,37 @@ If the built-in Kedro runners do not meet your requirements, you can also define
 ??? example "View code"
     ```python
     # in src/<package_name>/runner.py
-    from typing import Any, Dict
-    from kedro.io import AbstractDataset, KedroDataCatalog, MemoryDataset
+    from kedro.io import DataCatalog
     from kedro.pipeline import Pipeline
     from kedro.runner.runner import AbstractRunner
     from pluggy import PluginManager
-
+    from concurrent.futures import Executor
 
     class DryRunner(AbstractRunner):
         """``DryRunner`` is an ``AbstractRunner`` implementation. It can be used to list which
         nodes would be run without actually executing anything. It also checks if all the
         necessary data exists.
         """
 
-        def __init__(self, is_async: bool = False, extra_dataset_patterns: Dict[str, Dict[str, Any]] = None):
+        def __init__(self, is_async: bool = False):
             """Instantiates the runner class.
 
             Args:
                 is_async: If True, the node inputs and outputs are loaded and saved
                     asynchronously with threads. Defaults to False.
-                extra_dataset_patterns: Extra dataset factory patterns to be added to the KedroDataCatalog
-                    during the run. This is used to set the default datasets.
             """
-            default_dataset_pattern = {"{default}": {"type": "MemoryDataset"}}
-            self._extra_dataset_patterns = extra_dataset_patterns or default_dataset_pattern
-            super().__init__(is_async=is_async, extra_dataset_patterns=self._extra_dataset_patterns)
+            super().__init__(is_async=is_async)
+
+        def _get_executor(self, max_workers: int) -> Executor | None:
+            """Method to provide the correct executor (e.g., ThreadPoolExecutor,
+            ProcessPoolExecutor or None if running sequentially).
+            """
+            return None
```

Comment:
You can just do
```
def _get_executor(self, max_workers: int) -> None:
        return None
```

That way you don't need to import the `Executor` for just the type checking.